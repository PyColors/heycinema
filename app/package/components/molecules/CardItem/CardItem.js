import React from 'react';
import PropTypes from 'prop-types';

const CardItem = ({ item }) => (
  <div className="card-item-wrapper">
    <li className="card-item-wrapper__card-item">{item}</li>
  </div>
);

CardItem.propTypes = {
  item: PropTypes.any
};

export default CardItem;
