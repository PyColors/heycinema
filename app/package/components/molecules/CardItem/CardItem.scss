.card-item-wrapper {
  &__card-item {
    list-style-type: none;
  }
}
