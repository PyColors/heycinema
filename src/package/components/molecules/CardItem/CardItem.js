import React from 'react';
// muicss
import Panel from 'muicss/lib/react/panel';
import Row from 'muicss/lib/react/row';
import Col from 'muicss/lib/react/col';

const CardItem = ({ movie }) => (

  <Panel className="card">
  <article className='article__card'>
  <Row>
    <Col xs="12" md="12">
    {movie.Title ? null : <p className="card__information__heading"><br/>No result has been found, try again</p>}
    </Col>

    <Col xs="12" md="4">
      {
        (movie.Poster)
          ? <div className='card__picture'> <img src={ movie.Poster } /> </div>
          : null
      }
    </Col>
    
    <Col xs="12" md="6">
    <div className='card__information'>
      <div>
        <h2 className="card__information__heading"> {movie.Title ? movie.Title : null}</h2>
        <h3 className="card__information__subtitle"> {movie.Released ? movie.Released : null}</h3>
          {
          (movie.Title) ? 
          <div>
            {movie.Ratings.map((rating) => {
              return (<p key={rating.Source} className="card__information__callout"><b>{rating.Source}</b>: <i>{rating.Value}</i></p>);
            })}
         </div>
        : null }
      </div>
    </div>
    </Col>
  </Row>
  </article>
  </Panel>
)

export default CardItem;
