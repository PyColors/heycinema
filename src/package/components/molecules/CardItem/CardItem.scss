.card {
  padding: 0;
  border-radius: $radius-button;
  animation: animationFrames ease 1s;

  &__article {
    height: 197px;
  }

  &__picture {
    height: 197px;
    position: relative;

    &:after {
      position: absolute;
      top: 0;
      right: -34px;
      content: "";
      height: 0px;
      width: 82px;
      border-bottom: 197px solid $white;
      border-left: 40px solid transparent;
      border-right: 40px solid transparent;

      @media only screen and (max-width: 768px) {
        content: inherit;
        border-bottom: inherit;
      }
    }

      img {
      display: block;
      height: 197px;
      width: 61%;
      border-top-left-radius: $radius-button;
      border-bottom-left-radius: $radius-button;

      @media only screen and (max-width: 768px) {
        width: inherit;
        margin: 15px auto 0;
        border-radius: $radius-button;
      }
    }
  }

  &__information {
    position: relative;
    z-index: 10;
    margin: 0 0 0 15px;
    text-align: left;

    @media only screen and (max-width: 768px) {
      margin: 0;
      text-align: center;
    }

    &__heading {
      font-family: Arial, Helvetica, sans-serif;
      font-weight: bold;
      font-size: 24px;
      color: $primary-color;
    }

    &__subtitle {
      margin: 0 0 8px;
      color: $tertiary-color;
      font-family: Arial, Helvetica, sans-serif;
    }

    &__callout {
      font-family: Arial, Helvetica, sans-serif;
      font-size: 15px;
      color: $secondary-color;
    }
  }

}

@keyframes animationFrames {
  0% {
    opacity:0;
    transform:  translate(0px,-25px)  ;
  }
  100% {
    opacity:1;
    transform:  translate(0px,0px);
  }
}
