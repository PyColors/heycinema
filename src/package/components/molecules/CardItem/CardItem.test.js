import React from 'react';
import { shallow } from 'enzyme';
import CardItem from './CardItem';

describe('CardItem', () => {

  const props = {
    movie: 'Poster'
  }

  let propsData = {
    movie: {
      Title: "Titanic",
      Released: "19 Dec 1997",
      Poster: "https://m.media-amazon.com/images/M/MV5BMDdmZGU3NDQtY2E5My00ZTliLWIzOTUtMTY4ZGI1YjdiNjk3XkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SX300.jpg",
      Ratings: [
        {
          Source: "Internet Movie Database",
          Value: "7.8/10"
        },
        {
          Source: "Rotten Tomatoes",
          Value: "89%"
        },
        {
          Source: "Metacritic",
          Value: "75/100"
        }
      ],
    }
  }

  it('should render correctly with no props', () => {
    const wrapper = shallow(<CardItem {...props} />);
    expect(wrapper.length).toEqual(1)
  });


  it('should render correctly CardItem with data', () => {
    const wrapper = shallow(<CardItem {...propsData} />);
    expect(wrapper.length).toEqual(1)
    expect(wrapper).toMatchSnapshot();
  });

});