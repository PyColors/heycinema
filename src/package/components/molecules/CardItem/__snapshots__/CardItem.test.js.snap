// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem should render correctly CardItem with data 1`] = `
<Panel
  className="card"
>
  <article
    className="article__card"
  >
    <Row
      className=""
    >
      <Col
        className=""
        lg={null}
        lg-offset={null}
        md="12"
        md-offset={null}
        sm={null}
        sm-offset={null}
        xl={null}
        xl-offset={null}
        xs="12"
        xs-offset={null}
      />
      <Col
        className=""
        lg={null}
        lg-offset={null}
        md="4"
        md-offset={null}
        sm={null}
        sm-offset={null}
        xl={null}
        xl-offset={null}
        xs="12"
        xs-offset={null}
      >
        <div
          className="card__picture"
        >
           
          <img
            src="https://m.media-amazon.com/images/M/MV5BMDdmZGU3NDQtY2E5My00ZTliLWIzOTUtMTY4ZGI1YjdiNjk3XkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SX300.jpg"
          />
           
        </div>
      </Col>
      <Col
        className=""
        lg={null}
        lg-offset={null}
        md="6"
        md-offset={null}
        sm={null}
        sm-offset={null}
        xl={null}
        xl-offset={null}
        xs="12"
        xs-offset={null}
      >
        <div
          className="card__information"
        >
          <div>
            <h2
              className="card__information__heading"
            >
               
              Titanic
            </h2>
            <h3
              className="card__information__subtitle"
            >
               
              19 Dec 1997
            </h3>
            <div>
              <p
                className="card__information__callout"
                key="Internet Movie Database"
              >
                <b>
                  Internet Movie Database
                </b>
                : 
                <i>
                  7.8/10
                </i>
              </p>
              <p
                className="card__information__callout"
                key="Rotten Tomatoes"
              >
                <b>
                  Rotten Tomatoes
                </b>
                : 
                <i>
                  89%
                </i>
              </p>
              <p
                className="card__information__callout"
                key="Metacritic"
              >
                <b>
                  Metacritic
                </b>
                : 
                <i>
                  75/100
                </i>
              </p>
            </div>
          </div>
        </div>
      </Col>
    </Row>
  </article>
</Panel>
`;
